### ISSUE TITLE
Form generator duplicate field names possible

### ISSUE THREAD

**Issue by bennyborn** [contao/contao#1440]
**Affected version(s)**
4.9.0

**Description**
When adding fields to a form there seems to be no check if the given field name is already in use. While the frontend does prevent showing multiple fields with the same name, the backend does not show any error message whatsoever.

**How to reproduce**
1. Create a new form
2. Add a field with name "foo"
3. Add another field with name "foo"
4. Only the last field with the same name will be visible in FE, the other(s) will be ignored

**Comment by aschempp** [comment]
having the same field name is imho a supported use case. An example would be multiple checkbox fields with name `foo[]`. It might also be supported for GET forms.

**Comment by bennyborn** [comment]
Agreed. But at least the backend should throw an error if the field name does not contain square brackets and is already taken. And there is still the problem that fields with the same name are only visible once.

**Comment by aschempp** [comment]
Reopening because https://github.com/contao/contao/pull/1607 was not merged
